data-collection/data-collection/View Controllers/Floating Panel View Controller/FloatingPanelController.swift

```diff
@@ -757,13 +757,17 @@ fileprivate class FloatingPanelNavigationController: UINavigationController {
     }
 }
 
-/// Protocol designed to be implemented by objects that will display a floating panel.
-protocol FloatingPanelPresenter: class {
-    /// The floating panel to be displayed.
-    var floatingPanelController: FloatingPanelController? { get set }
-}
+/// A convenience protocol that can be implemented by view controllers
+/// displaying a floating panel to easily present and dismiss the panel.
+protocol FloatingPanelPresenter: UIViewController { }
+
+extension FloatingPanelPresenter {
```

Comment:
I think these changes are great. With them, however, it appears the need for `FloatingPanelPresenter` has disappeared. The pattern established here is very nice and is conducive such that I think we could restore this tool for use with any `UIViewController` pretty easily.

```suggestion
extension UIViewController {
```

(If you take this suggestion, don't forget to remove the `MapViewController`'s adherence to `FloatingPanelPresenter`)

Reply:
@esreli 's suggestion makes sense.